SSZipArchive/SSZipArchive.m

```diff
@@ -50,27 +50,30 @@ + (BOOL)isFilePasswordProtectedAtPath:(NSString *)path {
         return NO;
     }
     
+    BOOL passwordProtected = NO;
     int ret = unzGoToFirstFile(zip);
     if (ret == UNZ_OK) {
         do {
             ret = unzOpenCurrentFile(zip);
             if (ret != UNZ_OK) {
-                return NO;
+                break;
             }
             unz_file_info fileInfo = {};
             ret = unzGetCurrentFileInfo(zip, &fileInfo, NULL, 0, NULL, 0, NULL, 0);
             if (ret != UNZ_OK) {
-                return NO;
+                break;
             } else if ((fileInfo.flag & 1) == 1) {
-                return YES;
+                passwordProtected = YES;
+                break;
             }
             
             unzCloseCurrentFile(zip);
             ret = unzGoToNextFile(zip);
         } while (ret == UNZ_OK);
     }
-    
-    return NO;
+
+    unzClose(zip);
```

Comment:
@Coeur I think unzClose is OK here. This function is "self-contained" - it opens a file and determines if it is password-protected.